### PR TITLE
Tweaks some instances of get_safe_turf so things like the nuclear disk doesn't accidentally teleport to the Icebox Syndicate Base

### PR DESCRIPTION
**Pull request by BurgerLUA**
Yes, I know preventing the nuke disk teleporting to the icebox syndicate base (or possibly the wendigo arena) is removing soul. Please don't kill me :(

## About The Pull Request

Adds some missing variables to instances of get_safe_turf() so they will only teleport to the given z-level.
Replaces some instances of get_safe_turf() with get_safe_random_station_turf().

## Why It's Good For The Game

First, the differences between get_safe_turf() and get_safe_random_station_turf():

### get_safe_turf()
- gets a random safe turf on a z-level (usually any of the staiton z-levels), not accounting for the /area/
- should be used if you don't care if it spawns on the station or not, or if you need to specifically teleport to a z-level
- not very expensive performance wise

###  get_safe_random_station_turf()
- gets a random safe turf that will always be a station area, ignoring z-level.
- should be used if you NEED the turf to be on a station's area
- slightly more expensive performance wise than get_safe_turf, but still very cheap

Some code was using get_safe_turf() when it should've been using get_safe_random_station_turf(), and some code that should be using get_safe_turf() were incorrectly using the zlevels arg instead of zlevel arg (Yes, there is a difference), or didn't include it at all.

All the changes were made to my best judgement. If you're curious about a change, please ask and I will explain why I did it.

## Changelog

:cl: BurgerBB
fix: Tweaks some instances of get_safe_turf so things like the nuclear disk doesn't accidentally teleport to the Icebox Syndicate Base
/:cl:

### CODE REVIEW

**Comment by BurgerLUA** [comment]
Funny, linters are failing on the gateway_test area because there are no station turfs for the nuke disk to teleport to.

**Review by Jacquerel — CHANGES_REQUESTED** [review]
I think you are a little too zealous with limiting things to their current Z level without a clear reason as to why.

File: code/modules/mapfluff/ruins/icemoonruin_code/hotsprings.dm [hunk 1]
```diff
@@ -36,6 +36,6 @@
 		addtimer(TRAIT_CALLBACK_REMOVE(transforming_mind, TRAIT_HOT_SPRING_CURSED, TRAIT_GENERIC), 10 SECONDS)
 		return
 
-	var/turf/return_turf = find_safe_turf(extended_safety_checks = TRUE, dense_atoms = FALSE)
+	var/turf/return_turf = find_safe_turf(zlevel = src.z, extended_safety_checks = TRUE, dense_atoms = FALSE)
```
[the remark this answers is not in the document]
Comment (BurgerLUA) [reply code/modules/mapfluff/ruins/icemoonruin_code/hotsprings.dm:39]: Ah, I didn't consider that. Will change back.

File: code/modules/mapfluff/ruins/spaceruin_code/hilbertshotel.dm [hunk 2]
```diff
@@ -425,7 +425,7 @@ GLOBAL_VAR_INIT(hhMysteryRoomNumber, rand(1, 999999))
 		unforeseen_consequences.investigate_log("has been gibbed by using [H] while inside of it.", INVESTIGATE_DEATHS)
 		unforeseen_consequences.gib(DROP_ALL_REMAINS)
 
-	var/turf/targetturf = find_safe_turf()
+	var/turf/targetturf = find_safe_turf(zlevel = src.z)
```
[the remark this answers is not in the document]
Comment (BurgerLUA) [reply code/modules/mapfluff/ruins/spaceruin_code/hilbertshotel.dm:428]: Will change.

File: code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm [hunk 3]
```diff
@@ -261,7 +261,7 @@ Difficulty: Extremely Hard
 	clone.real_name = user.real_name
 	INVOKE_ASYNC(user.dna, TYPE_PROC_REF(/datum/dna, transfer_identity), clone)
 	clone.updateappearance(mutcolor_update=1)
-	var/turf/T = find_safe_turf()
+	var/turf/T = find_safe_turf(zlevel = user.z, extended_safety_checks = TRUE)
```
[the remark this answers is not in the document]
Comment (BurgerLUA) [reply code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm:264]: I could implement that. The spawning logic just places you in a maint marker location.

File: code/modules/projectiles/guns/magic/wand.dm [hunk 4]
```diff
@@ -192,7 +192,7 @@
 
 /obj/item/gun/magic/wand/safety/zap_self(mob/living/user)
 	var/turf/origin = get_turf(user)
-	var/turf/destination = find_safe_turf()
+	var/turf/destination = find_safe_turf(zlevel = user.z, extended_safety_checks = TRUE)
```
[the remark this answers is not in the document]
Comment (BurgerLUA) [reply code/modules/projectiles/guns/magic/wand.dm:195]: It's mostly just a "lore" reason. There is no reason why a safety wand would stick to the station when it is possible (albeit rare) that ashwalkers could use it.

File: code/modules/projectiles/projectile/magic.dm [hunk 5]
```diff
@@ -110,7 +110,7 @@
 		return BULLET_ACT_HIT
 
 	var/turf/origin_turf = get_turf(target)
-	var/turf/destination_turf = find_safe_turf()
+	var/turf/destination_turf = find_safe_turf(zlevel = target.z, extended_safety_checks = TRUE)
```
[the remark this answers is not in the document]
Comment (BurgerLUA) [reply code/modules/projectiles/projectile/magic.dm:113]: Same as above.

**Review by Jacquerel — CHANGES_REQUESTED** [review]

File: code/modules/mapfluff/ruins/icemoonruin_code/hotsprings.dm [hunk 6]
```diff
@@ -36,6 +36,6 @@
 		addtimer(TRAIT_CALLBACK_REMOVE(transforming_mind, TRAIT_HOT_SPRING_CURSED, TRAIT_GENERIC), 10 SECONDS)
 		return
 
-	var/turf/return_turf = find_safe_turf(zlevel = src.z, extended_safety_checks = TRUE, dense_atoms = FALSE)
+	var/turf/return_turf = find_maintenance_spawn(atmos_sensitive = TRUE, require_darkness = TRUE) || find_safe_turf(extended_safety_checks = TRUE, dense_atoms = FALSE)
```
[the remark this answers is not in the document]
Comment (BurgerLUA) [reply code/modules/mapfluff/ruins/icemoonruin_code/hotsprings.dm:39]: oh god I meant to apply this to the demonic frost miner one lmao

File: code/modules/antagonists/wizard/equipment/artefact.dm [hunk 7]
```diff
@@ -378,7 +378,7 @@
 	addtimer(CALLBACK(src, PROC_REF(send_away)), 2 SECONDS)
 
 /obj/effect/temp_visual/teleporting_tornado/proc/send_away()
-	var/turf/ending_turfs = find_safe_turf()
+	var/turf/ending_turfs = find_safe_turf(zlevel = src.z)
```
Comment (Jacquerel) [inline code/modules/antagonists/wizard/equipment/artefact.dm:381]: Don't know how I missed this one but I think the warp whistle shouldn't be limited to the current z, although maybe station turf would be fine? The description says "Calls a cloud to come pick you up and drop you at a random location on the station".

**Review by Jacquerel — CHANGES_REQUESTED** [review]
One more thing I noticed then we should be good

File: code/modules/antagonists/wizard/equipment/artefact.dm (hunk 7 above)
Comment (BurgerLUA) [reply code/modules/antagonists/wizard/equipment/artefact.dm:381]: will do

**Pull request merged**